packages/utilities/src/styled.tsx

```diff
@@ -59,79 +64,85 @@ export function styled<
   getProps?: (props: TComponentProps) => Partial<TComponentProps>,
   customizable?: ICustomizableProps,
   pure?: boolean,
-): React.FunctionComponent<TComponentProps> {
+): React.FunctionComponent<TComponentProps>;
+export function styled<
+  TRef,
+  TComponentProps extends IPropsWithStyles<TStyleProps, TStyleSet> & React.RefAttributes<TRef>,
+  TStyleProps,
+  TStyleSet extends IStyleSet<TStyleSet>
+>(
+  Component: React.ComponentClass<TComponentProps> | React.FunctionComponent<TComponentProps>,
+  baseStyles: IStyleFunctionOrObject<TStyleProps, TStyleSet>,
+  getProps?: (props: TComponentProps) => Partial<TComponentProps>,
+  customizable?: ICustomizableProps,
+  pure?: boolean,
+): React.ForwardRefExoticComponent<React.PropsWithoutRef<TComponentProps> & React.RefAttributes<TRef>>;
+export function styled<
+  TRef,
+  TComponentProps extends IPropsWithStyles<TStyleProps, TStyleSet> & React.RefAttributes<TRef>,
+  TStyleProps,
+  TStyleSet extends IStyleSet<TStyleSet>
+>(
+  Component: React.ComponentClass<TComponentProps> | React.FunctionComponent<TComponentProps>,
+  baseStyles: IStyleFunctionOrObject<TStyleProps, TStyleSet>,
+  getProps?: (props: TComponentProps) => Partial<TComponentProps>,
+  customizable?: ICustomizableProps,
+  pure?: boolean,
+) {
   customizable = customizable || { scope: '', fields: undefined };
 
   const { scope, fields = DefaultFields } = customizable;
-  const ParentComponent = pure ? React.PureComponent : React.Component;
 
-  class Wrapped extends ParentComponent<TComponentProps, {}> {
-    // Function.prototype.name is an ES6 feature, so the cast to any is required until we're
-    // able to drop IE 11 support and compile with ES6 libs
-    // tslint:disable-next-line:no-any
-    public static displayName = `Styled${Component.displayName || (Component as any).name}`;
+  const Wrapped = React.forwardRef((props: TComponentProps, forwardedRef: React.Ref<TRef>) => {
```

Comment:
i dont think fowardRef is needed

Reply:
The entire impetus for this change was the fact that I *do* need to forward a ref across a `styled` boundary.

Reply:
icic